pkg/investigations/cannotretrieveupdatessre/cannotretrieveupdatessre.go

```diff
@@ -0,0 +1,124 @@
+		switch verifierResult {
+		case networkverifier.Failure:
+			logging.Infof("Network verifier reported failure: %s", failureReason)
+			// XXX: metrics.Inc(metrics.ServicelogPrepared, investigationName)
+			result.ServiceLogPrepared = investigation.InvestigationStep{Performed: true, Labels: nil}
+			notes.AppendWarning("NetworkVerifier found unreachable targets. \n \n Verify and send service log if necessary: \n osdctl servicelog post %s -t https://raw.githubusercontent.com/openshift/managed-notifications/master/osd/required_network_egresses_are_blocked.json -p URLS=%s", r.Cluster.ID(), failureReason)
+		case networkverifier.Success:
+			notes.AppendSuccess("Network verifier passed")
+		}
+	}
+
+	// Check ClusterVersion
+	clusterVersion, note, err := checkClusterVersion(k8scli, r.Cluster.ID())
+	if err != nil {
+		notes.AppendWarning("Failure checking ClusterVersion: %s", err.Error())
+	} else {
+		if note != "" {
+			notes.AppendWarning(note)
+		}
+		if clusterVersion != "" {
+			notes.AppendSuccess("ClusterVersion found: %s", clusterVersion)
+		}
+	}
+
+	notes.AppendWarning("Alert escalated to on-call primary for review.")
+	logging.Infof("Escalating incident with notes for cluster %s", r.Cluster.ID())
+	err = r.PdClient.EscalateIncidentWithNote(notes.String())
+	if err != nil {
+		logging.Errorf("Failed to escalate incident to PagerDuty: %v", err)
+		return result, fmt.Errorf("failed to escalate incident: %w", err)
+	}
+	logging.Infof("Investigation completed and escalated successfully for cluster %s", r.Cluster.ID())
+
+	return result, nil
+}
+
+// checkClusterVersion retrieves the cluster version
+func checkClusterVersion(k8scli client.Client, clusterID string) (version string, note string, err error) {
+	logging.Infof("Checking ClusterVersion for cluster %s", clusterID)
+	clusterVersion := &configv1.ClusterVersion{}
+	err = k8scli.Get(context.TODO(), client.ObjectKey{Name: "version"}, clusterVersion)
+	if err != nil {
+		return "", "Failed to get ClusterVersion: cluster access issues detected", fmt.Errorf("failed to get ClusterVersion: %w", err)
+	}
+	logging.Infof("ClusterVersion channel: %s", clusterVersion.Spec.Channel)
+	logging.Infof("ClusterVersion found: %s", clusterVersion.Status.Desired.Version)
+
+	for _, condition := range clusterVersion.Status.Conditions {
+		if condition.Type == "RetrievedUpdates" && condition.Status == "False" {
+			if (condition.Reason == "VersionNotFound" || condition.Reason == "RemoteFailed") &&
+				strings.Contains(strings.TrimSpace(condition.Message), "Unable to retrieve available updates") {
+				logging.Warnf("Detected ClusterVersion issue: Reason=%s, Message=%s", condition.Reason, condition.Message)
+				return "", fmt.Sprintf("ClusterVersion issue detected: %s. Current version %s not found in channel %s",
+						condition.Message, clusterVersion.Status.Desired.Version, clusterVersion.Spec.Channel),
+					fmt.Errorf("clusterversion has undesirable state: %s", condition.Reason)
+			}
+		}
+	}
```

Comment:
```suggestion
	for _, condition := range clusterVersion.Status.Conditions {
		if condition.Type == "RetrievedUpdates" {
		    note, err := checkRetrievedUpdatesCondition(condition)
		    return clusterVersion.Status.DesiredVersion, note, err
		}
	}
```
It looks like we're only looking to analyze the state of one condition here, correct? Would it make more sense to break some of this out into it's own function? 

Something like the following may make the nested conditionals more readable:

```
func checkRetrievedUpdatesCondition(condition corev1.StatusCondition) (string, err) {
    if condition.Status == corev1.ConditionFalse {
        return ...
    }
    if (condition.Reason == ...) {
        return ...
    }
}
```

Additionally, we won't waste CPU cycles continuing to loop through conditions we don't need to analyze once we find the one we're after.